src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -159,31 +176,67 @@ else if (length == 3 && !hasMs || length == 4) // HH:MM:SS[.ms]
 					minecraftTime = true;
 					sub = substring[++i];
 				}
-				
+
 				if (sub.endsWith(","))
 					sub = sub.substring(0, sub.length() - 1);
 
-				Long d = PARSE_VALUES.get(sub.toLowerCase(Locale.ENGLISH));
-				if (d == null)
+				Long millis = PARSE_VALUES.get(sub.toLowerCase(Locale.ENGLISH));
+				if (millis == null)
 					return null;
-				
-				if (minecraftTime && d != TimePeriod.TICK.time)
+
+				if (minecraftTime && millis != TimePeriod.TICK.time)
 					amount /= 72f;
-				
-				t += Math.round(amount * d);
-				
+
+				totalMillis += Math.round(amount * millis);
+
 				isMinecraftTimeSet = true;
-				
 			}
 		}
 
-		return new Timespan(t);
+		return totalMillis > 0 ? new Timespan(totalMillis) : null;
 	}
-	
+
 	public Timespan() {
 		millis = 0;
 	}
 
+	/**
+	 * Parses a string representing a shortened timespan, such as "1d" for 1 day,
+	 * "2h" for 2 hours, or "30m" for 30 minutes.
+	 * @param value The string representing the shortened timespan.
+	 * @return A {@link Timespan} object representing the parsed timespan, or {@code null} if the input string is invalid.
+	 */
+	@Nullable
+	public static Timespan parseShortenedTimespan(String value) {
+		if (value.isEmpty()) {
+			return null;
+		}
+
+		long totalMillis = 0;
+		Pattern pattern = Pattern.compile("(\\d+)([a-zA-Z]+)");
```

Comment:
This should be a field since it doesn't need to be done multiple times.

Since we have all of the short forms of the time periods, you could actually just make a regex searching for those rather than any combination of random letters.